# Thesis-R-scripts
BENEFICIAL PLANTË—MICROBE INTERACTION IN AGROECOSYSTEMS: DECIPHERING THE RHIZOSPHERE MICROBIAL COMMUNITY IN FIELD GROWN Brassica napus L.

This repository contain codes related to my PhD thesis "Beneficial Plant-microbe Interactions in Agroecosystems: Deciphering the Rhizosphere Microbial Community in Field Grown Brassica napus L." presented for the degree of Doctor of Philosophy at the University of Saskatchewan in 2020. 

## Codes used
The scripts are being formated in order to make them more readily usable. I will be updating this repository as this progresses, and this sentence will be removed once all scripts used have been uploaded accordingly.  

## Abstract

Understanding the microbiome and factors governing its structure under field conditions is essential to optimize the microbiome for improved crop productivity. The purpose of my research was to characterize the rhizosphere bacterial community of sixteen diverse Brassica napus lines under field conditions to identify microbial patterns and specific signatures that could be utilized in canola breeding and production management strategies.

My first objective was to characterize the core rhizosphere microbiome and identify components of the bacterial microbiota that differ between B. napus genotypes. There were both cross-year and cross-site conserved core bacterial genera. There was a fine-scale plant genetic control on rhizosphere bacterial abundance, and plant genetics explained up to 59% of the variation in bacterial alpha diversity. Overall microbial variation and plant genetic distance were correlated demonstrating plant control on the rhizosphere ecology. Moreover, the plant genetic effects on the microbial community varied in strength through plant development with a maximum at flowering.

My second objective was to characterize B. napus root growth dynamics and assess associations between root growth and dominant bacterial taxa. Changes in the dominant bacteria were detectable at all levels of taxonomic resolution through plant development. Root growth showed a distinct growth stage pattern with an increase in root length at early stages followed by stable and/or gradual increases at flowering and reductions at maturity. The abundance of dominant bacteria showed significant positive correlations with root length in vegetative plants.

My third objective was to characterize microbial interactions and identify driver and core-hub bacterial communities structuring B. napus microbial assembly. Microbial network structure varied across growth stages with more complex and organized structures at flowering. Potential core-hub communities and driver taxa influencing microbial assembly processes and the metabolic pathways of the driver genera were identified. Overall, the B. napus rhizosphere microbial assembly is characterized by conserved bacterial interactions (core-hub communities) and altered bacterial interactions between growth stages or genotypes that create variation (driver genera).

My research has filled a significant knowledge gap in B. napus microbiome literature by identifying the overall, growth stage and genotype related microbial patterns and signatures necessary for future detailed investigations. These novel insights into the rhizosphere bacterial community will enable the design of future strategies that combine both biocontrol and breeding approaches to address challenges in B. napus production. 

